lib/core/test/unit/Data/Vector/ShuffleSpec.hs

```diff
@@ -54,3 +74,43 @@ prop_shufflePreserveElements
 prop_shufflePreserveElements xs = monadicIO $ run $ do
     xs' <- shuffle xs
     return $ cover 90 (not $ null xs) "non-empty" (L.sort xs == L.sort xs')
+
+-- ∀(g :: RandomGen).
+-- ∀(es :: [a]).
+--
+-- shuffleWith g es == shuffleWith g es
+prop_shuffleWithDeterministic
+    :: PrintableString
+    -> NonEmptyList Int
+    -> Property
+prop_shuffleWithDeterministic (PrintableString seed) (NonEmpty xs) =
+    monadicIO $ do
+        ys0 <- run $ shuffleWith (mkSeed $ T.pack seed) xs
+        ys1 <- run $ shuffleWith (mkSeed $ T.pack seed) xs
+        monitor $ cover 90 (length xs > 1) "non singleton"
+        assert (ys0 == ys1)
+
+-- ∀(x0 : Text, x1 : Text). g0 = mkSeed x0, g1 = mkSeed x1
+-- ∃(Δ: Int).
+-- ∀(es :: [a]).
+--
+-- g0 ≠g1, length es > Δ⇒ shuffleWith g0 es ≠shuffleWith g1 es
```

Comment:
Nit-pick / confusion: the `∃(Δ: Int)` part sounds 1) questionable to me, and 2) not the same as the code?. Whatever `∆`, given enough time, this property is always disprovable?

Reply:
Well, to be honest, I was hoping to have the property to fail and have to find a threshold manually (even though this is entirely subjective to the QC generator).
 
Still, it's hard to translate in pure maths here because that's really a probabilistic results. We can't possibly guarantee that this is true for any list and any seed. But, _most of the time_, it should be a reasonable assumption. In particular, when lists are big enough (and it's this is what the threshold translates), it should hold with a reasonable confidence. 

Turns out that, 90% of the results satisfies the property already with a threshold of `0` :man_shrugging:, QC isn't generating that many singletons.